wp_api/src/uniffi_serde.rs

```diff
@@ -0,0 +1,30 @@
+#[derive(Debug, uniffi::Object, thiserror::Error)]
```

Comment:
As far as I know, we use `uniffi::Error` type for all other error types. Could you elaborate why we should use `uniffi::Object` here? There is no exported function to read the error details at the moment, so I don't know what the intention is.

Even if we need to wrap `serde_json::Error` type, I think my preference would be for us to use `uniffi::Error` for the error type itself and put the `uniffi::Object` into one of the variants. That way we can also add some of the expected error variants as the `enum` variant.

Also, I vaguely remember not using `uniffi::Error` in `Result<T, E>` causing issues with `uniffi` - possibly in Kotlin? I unfortunately can't remember the details and maybe they have changed recently.

Reply:
I just thought it'd be a simpler solution than a one-variant enum.

Reply:
I have changed it to an enum in https://github.com/Automattic/wordpress-rs/pull/559/commits/c93a8c73730154f80540870a27518624124484a1